doc/java_parser.md

```diff
@@ -0,0 +1,81 @@
+---
+title: JavaParser 
+keywords: JavaParser
+---
+
+Here is JavaParser github project: [JavaPaser](https://github.com/javaparser/javaparser)
+
+## Differences between JavaParser and Spoon
+
+Spoon and [JavaParser](https://github.com/javaparser/javaparser) can both be used for analyzing and transforming Java source code.
+
+### Parsing
+
+Spoon is using Eclipse JDK compiler to build AST of sources, then Spoon AST is then made from compiler AST.
+JavaParser parses java source on their own. So JavaParser is faster, but all the heavy work (resolving of fully qualified type names, detection of implicit elements, ...) is done by JavaParser (@ftomassetti could you add there some comment about status)

```

Comment:
JavaParser can work incrementally. It can just parse, using its own parser. It can work on snippets or entire files. Optionally it can also perform symbol resolution, which again it is implemented from scratch and it is very flexible.